core/drivers/versal_huk.c

```diff
@@ -0,0 +1,339 @@
+
+		versal_mbox_alloc(key_len, key_data, &p);
+		cmd.data[0] = API_ID(VERSAL_AES_WRITE_KEY);
+		cmd.data[1] = VERSAL_AES_KEY_SIZE_256;
+		cmd.data[2] = key_id;
+		reg_pair_from_64(virt_to_phys(p.buf),
+				 &cmd.data[4], &cmd.data[3]);
+		cmd.ibuf[0].mem = p;
+		if (versal_mbox_notify(&cmd, NULL, NULL)) {
+			EMSG("AES_WRITE_KEY error");
+			ret = TEE_ERROR_GENERIC;
+		}
+		free(p.buf);
+		memset(&cmd, 0, sizeof(cmd));
+		if (ret)
+			return ret;
+	}
+
+	/* Trace indication that it is safe to generate a RPMB key */
+	IMSG("Using %s HUK", secure ? "Production" : "Development");
+
+	versal_mbox_alloc(sizeof(*init), NULL, &init_buf);
+	versal_mbox_alloc(nce_len, nce_data, &p);
+	init = init_buf.buf;
+	init->operation = VERSAL_AES_GCM_ENCRYPT;
+	init->key_len = VERSAL_AES_KEY_SIZE_256;
+	init->iv_addr = virt_to_phys(p.buf);
+	init->key_src = key_id;
+	cmd.data[0] = API_ID(VERSAL_AES_OP_INIT);
+	reg_pair_from_64(virt_to_phys(init), &cmd.data[2], &cmd.data[1]);
+	cmd.ibuf[0].mem = init_buf;
+	cmd.ibuf[1].mem = p;
+	if (versal_mbox_notify(&cmd, NULL, NULL)) {
+		EMSG("AES_OP_INIT error");
+		ret = TEE_ERROR_GENERIC;
+	}
+	free(init);
+	free(p.buf);
+	memset(&cmd, 0, sizeof(cmd));
+	if (ret)
+		return ret;
+
+	versal_mbox_alloc(aad_len, aad_data, &p);
+	cmd.data[0] = API_ID(VERSAL_AES_UPDATE_AAD);
+	reg_pair_from_64(virt_to_phys(p.buf), &cmd.data[2], &cmd.data[1]);
+	if (p.len % 16)
+		cmd.data[3] = p.alloc_len;
+	else
+		cmd.data[3] = p.len;
+	cmd.ibuf[0].mem = p;
+	if (versal_mbox_notify(&cmd, NULL, NULL)) {
+		EMSG("AES_UPDATE_AAD error");
+		ret = TEE_ERROR_GENERIC;
+	}
+	free(p.buf);
+	memset(&cmd, 0, sizeof(cmd));
+	if (ret)
+		return ret;
+
+	versal_mbox_alloc(sizeof(*input), NULL, &input_cmd);
+	versal_mbox_alloc(src_len, src, &p);
+	versal_mbox_alloc(dst_len, NULL, &q);
+	input = input_cmd.buf;
+	input->input_addr = virt_to_phys(p.buf);
+	input->input_len = p.len;
+	input->is_last = true;
+	cmd.data[0] = API_ID(VERSAL_AES_ENCRYPT_UPDATE);
+	reg_pair_from_64(virt_to_phys(input), &cmd.data[2], &cmd.data[1]);
+	reg_pair_from_64(virt_to_phys(q.buf), &cmd.data[4], &cmd.data[3]);
+	cmd.ibuf[0].mem = input_cmd;
+	cmd.ibuf[1].mem = p;
+	cmd.ibuf[2].mem = q;
+	if (versal_mbox_notify(&cmd, NULL, NULL)) {
+		EMSG("AES_UPDATE_PAYLOAD error");
+		ret = TEE_ERROR_GENERIC;
+	}
+	memcpy(dst, q.buf, dst_len);
+	free(input);
+	free(p.buf);
+	free(q.buf);
+	memset(&cmd, 0, sizeof(cmd));
+	if (ret)
+		return ret;
+
+	versal_mbox_alloc(16, NULL, &p);
+	cmd.data[0] = API_ID(VERSAL_AES_ENCRYPT_FINAL);
+	reg_pair_from_64(virt_to_phys(p.buf), &cmd.data[2], &cmd.data[1]);
+	if (versal_mbox_notify(&cmd, NULL, NULL)) {
+		EMSG("AES_ENCRYPT_FINAL error");
+		ret = TEE_ERROR_GENERIC;
+	}
+	free(p.buf);
+	memset(&cmd, 0, sizeof(cmd));
+
+	return ret;
+}
+
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	uint32_t dna[EFUSE_DNA_LEN / sizeof(uint32_t)] = { 0 };
+	uint8_t enc_data[64] = { 0 };
+	uint8_t sha[48] = { 0 };
```

Comment:
`= { };`

Reply:
`= { };` is quite clear too and having only one option is simpler...

Reply:
ok, let's sitck to the conding standards. I'll fix the stm32 occurences after next tag, for consistency.